docker/docker_client.go

```diff
@@ -419,6 +404,30 @@ func (c *dockerClient) makeRequest(ctx context.Context, method, path string, hea
 // makeRequest should generally be preferred.
 // TODO(runcom): too many arguments here, use a struct
 func (c *dockerClient) makeRequestToResolvedURL(ctx context.Context, method, url string, headers map[string][]string, stream io.Reader, streamLen int64, auth sendAuth, extraScope *authScope) (*http.Response, error) {
+	var (
+		res   *http.Response
+		err   error
+		delay time.Duration
+	)
+	// Do an exponential back off starting at 2 seconds for at most 5
+	// iterations.
+	delay = 2
+	for i := 0; i < 5; i++ {
+		res, err = c.makeRequestToResolvedURLHelper(ctx, method, url, headers, stream, streamLen, auth, extraScope)
```

Comment:
`stream` can be only consumed once; this fundamentally can’t be done, at least not at this level of abstraction.

Reply:
Good point. Looking at the call sites, we should not perform any retry when `stream != nil` which avoids the issue.